src/common.cpp

```diff
@@ -1873,6 +1880,58 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
+#elif defined(__OpenBSD__)
+    static std::string result;
+    if (!result.empty()) return result;
+    std::string strargv0 = ((argv0) ? argv0 : ""), path;
+    struct stat st;
+    if (!strargv0.empty()) {
+        if (strargv0[0] == '/') {
+            path = strargv0;
+        } else if (path.find('/') == std::string::npos) {
+            const char *penv = getenv("PATH");
+            if (penv && *penv) {
+                std::vector<std::string> env = split_string(penv, ':');
+                for (std::size_t i = 0; i < env.size(); i++) {
+                    path = env[i] + "/" + strargv0;
+                    if (!stat(path.c_str(), &st) && (st.st_mode & S_IXUSR) && (st.st_mode & S_IFREG)) {
+                        break;
+                    }
+                }
+            }
+        } else {
+            const char *pwd = getenv("PWD");
+            if (pwd && *pwd) {
+                path = std::string(pwd) + "/" + strargv0;
+            }
+        }
+    }
+    kvm_t *kd = kvm_openfiles(nullptr, nullptr, nullptr, KVM_NO_FILES, nullptr);
+    if (!kd) { result = "fish"; return result; }
+    int cntp = 0;
+    bool ok = false;
+    kinfo_file *kif = nullptr;
+    if ((kif = kvm_getfiles(kd, KERN_FILE_BYPID, getpid(), sizeof(struct kinfo_file), &cntp))) {
+        for (int i = 0; i < cntp; i++) {
+            if (kif[i].fd_fd == KERN_FILE_TEXT) {
+                if (!stat(path.c_str(), &st)) {
```

Comment:
Oh, I just noticed, you're stat'ing `path` over and over again.

Extract this `if` out of the for-loop and just keep `st` around.

Since `path` doesn't change here, if the `stat` fails you can just skip this entire thing and return `"fish"`.